node-bootstrapper/app.go

```diff
@@ -100,6 +104,60 @@ func (a *App) provisionStart(ctx context.Context, cse utils.SensitiveString) err
 	return err
 }
 
+// usage example:
+// node-bootstrapper monitor
+func (a *App) Monitor(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 15*time.Minute)
+	defer cancel()
+
+	for {
+		select {
+		case <-timeoutCtx.Done():
+			// If the timeout or cancel occurs, exit with a timeout error
+			return fmt.Errorf("monitoring timed out: %s still active after 15 minutes", BootstrapService)
+		default:
+			_, err := os.Stat(ProvisionJSONFilePath)
```

Comment:
am I right in understanding that once the unit is finished, it will write the status to this file path?

if yes, why do we need to check both existence of the file path and whether unit is active?

keep sleeping/retrying till unit is running, exit if failed and if not, then check file content? this check seems redundant unless i am missing something.

is there a delay between unit finishing vs writing to file?

Reply:
Yes im not sure why where seems to be a delay in `provision.json` being present and the systemd unit completing but without the additional file check, I ran into the case where the systemd unit finished but the file was not present.

Reply:
Changed to only waiting for `provision.json` since that is all we need to check bootstrap completion. If timeout occurs, return status of systemd unit.